src/sherpa_ai/config/task_config.py

```diff
@@ -50,5 +60,17 @@ def from_config(cls, configs: List[str]) -> "AgentConfig":
 
         if len(unknown) > 0:
             raise ValueError(f"Invalid configuration, check your input: {unknown}")
+        
+        if args.gsite:
+            gsite_list = [url.strip() for url in args.gsite.split(",")]
+            args.search_domains = [url for url in gsite_list if validate_url(url)]
```

Comment:
I'm thinking now, these attributes are derived from the `gsite` attribute. How does the `@computed_feature` and `@cached_property` can potentially help us. Maybe take a look at this link: https://docs.pydantic.dev/2.0/usage/computed_fields/. And let's discuss the findings.